wgpu/backends/wgpu_native/_api.py

```diff
@@ -827,14 +850,40 @@ def _request_device(
         def device_lost_callback(c_reason, c_message, userdata):
             reason = enum_int2str["DeviceLostReason"].get(c_reason, "Unknown")
             message = ffi.string(c_message).decode(errors="ignore")
-            error_handler.log_error(f"The WGPU device was lost ({reason}):\n{message}")
+            msg = f"The WGPU device was lost ({reason}):\n{message}"
+            # This is afaik an error that cannot usually be attributed to a specific call,
+            # so we cannot raise it as an error. We log it instead.
```

Comment:
I am often encountering something like 

```bash
thread '<unnamed>' panicked at src\lib.rs:582:5:
Error in wgpuQueueSubmit: Validation Error

Caused by:
  Parent device is lost

note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
where the python process crashes and is lost. As the message is logged and seen by stderr(I think), it would be great to handle this right here.
The message says Validation error, and for some CI tests for me locally is equivalent to the actually handled error.

Could this be an issue here the device lost callback is reached before the uncaptured error callback (which does handle the error and riseses and exception?).

Reply:
> I am often 

How often, and under what conditions?

Reply:
let me get up and reduce some of these fatal errors into minimal repros. Maybe adding them as a `test_fatal_errors.py`? I collected quite a few but it's in a messy private repo.

Reply:
On second thought, reading this from the WebGPU docs:
<img width="826" alt="image" src="https://github.com/user-attachments/assets/80526642-8669-41d3-a310-c48ee3b0b693">

I think the device getting lost is not supposed to result in an exception. It means the device is used in a bad way. The `device.lost` promise can be used to do something application-specific (but we have not implemented that). 

The fact that trying to use the device anyway, should be considered a bug in wgpu-core; it should result in an error (Python exception) instead.

Reply:
My earlier change to handle this as validation error was to address CI/pytest which had `buffer_init3` and `native_errors` throw a "did not raise exception" failure. But still log a message that looked like the one expected. 

See for example here https://github.com/pygfx/wgpu-py/actions/runs/10704356989/job/29677159712?pr=547
The root cause is odd to me however. Since these tests fail as expected (with `GPUValidationError` raise) when run in isolation or some subset... Couldn't figure out what combination of tests caused this - apart from the full suite

Reply:
I pulled your changes and the test suite now passes locally on Vulkan. (only the texture formats issue still fails)